Closes #312 - Create Transaction controller, service

hlibOstapenko opened the pull request.

hlibOstapenko committed:
Created service, controller, DTOs for transactions

hlibOstapenko committed:
Changed signature name in methods

hlibOstapenko committed:
Created methods for buying Decks and Courses.
Created TransactionType enum.
Updated flyway script v1.12.

hlibOstapenko committed:
Resolve conflicts

antonbakalets commented on src/main/java/com/softserve/academy/spaced/repetition/repository/PointsTransactionRepository.java:
```diff
@@ -2,6 +2,21 @@
+    @Query(value = "select distinct p from PointsTransaction p where p.userFrom.id = :idUser or p.userTo.id = :idUser")
+    List<PointsTransaction> findAllTransactionsByUserId(@Param("idUser") long idUser);
+
+    @Query(value = "select p from PointsTransaction p")
```
There is already a method JpaRepositoryfindAll()

hlibOstapenko replied:
Yeah, dummy me

antonbakalets commented on src/main/java/com/softserve/academy/spaced/repetition/utils/exceptions/TransactionException.java:
```diff
@@ -0,0 +1,11 @@
+package com.softserve.academy.spaced.repetition.utils.exceptions;
+
+public class TransactionException extends ApplicationException {
```
Please rename to PointsTransactionException, because now it looks like something related to database transaction.

hlibOstapenko committed:
Refactored PointsTransactionDTO.
Updated PointsTransactionRepository and Service implementation.
Deleted unused TransactionDTO.

hlibOstapenko committed:
Refactored PointsTransactionDTO.
Updated PointsTransactionRepository and Service implementation.
Deleted unused TransactionDTO.
